packages/ra-core/src/controller/list/useListController.ts

```diff
@@ -475,13 +523,15 @@ export interface ListControllerBaseResult<RecordType extends RaRecord = any> {
     filterValues: any;
     hideFilter: (filterName: string) => void;
     onSelect: (ids: RecordType['id'][]) => void;
+    onSelectAll: () => void;
     onToggleItem: (id: RecordType['id']) => void;
     onUnselectItems: () => void;
     page: number;
     perPage: number;
     refetch: (() => void) | UseGetListHookValue<RecordType>['refetch'];
     resource: string;
     selectedIds: RecordType['id'][];
+    displaySelectAllButton: boolean;
```

Comment:
The naming is too tied to UI concerns IMO. I would name this: areAllItemsSelected or similar